transformer_lens/HookedSAE.py

```diff
@@ -0,0 +1,95 @@
+from torch import nn
+import torch
+import torch.nn.functional as F
+import einops
+from transformer_lens.hook_points import (
+    HookedRootModule,
+    HookPoint,
+)  # Hooking utilities
+from transformer_lens.HookedSAEConfig import HookedSAEConfig
+
+from typing import Union, Dict
+
+
+class HookedSAE(HookedRootModule):
+    """Hooked AutoEncoder.
+    
+    Implements a standard SAE with a TransformerLens hooks for SAE activations
+    
+    Should probably just be used for inference / analysis, not training.
```

Comment:
Add link to Joseph's codebase for training.

Reply:
FWIW, I'd love to make my codebase also an inference codebase and build on this. Having this in T-lens seems good but I think will lead to lots of changes to adapt to new SAE methodologies which will create lots of work in T-lens which already has a lot going on. I think have this PR here accepted is probably good but in the long run, there should be other SAE analysis code bases.